### PR TITLE
[docs] Write markdown blocks with explanations for fft-net notebook

### PR DESCRIPTION
**Pull request by ninakuhlen**

### CODE REVIEW

**Comment by ninakuhlen** [comment]
before merging check if outputs are cleared

File: notebooks/fft_net.ipynb [hunk 1]
```diff
@@ -149,11 +197,22 @@
...
     "}"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In der Funktion `construct_fft_net_model` wird die *Keras Sequential API* eingesetzt, um ein NN als linearen Stapel von Layern zu definieren (Keras, o.D.-c). Hierfür wird eine Instanz der Klasse Sequential erzeugt und mit Input-, Output- sowie der zuvor spezifizierten Anzahl Hidden Layers erweitert.\n",
+    "\n",
+    "Je nach Eingabeformat der Daten (1D-Struktur oder sequentielle Daten), wird dynamisch entschieden, ob eine *Fully Connected* oder eine *Convolutional* Netzwerkarchitektur erstellt wird. Im Fall eines Fully Connected Netzes werden, analog zu Mey et al (2020), 2048 Neuronen in der ersten Dense-Schicht verwendet und durch *L2-Regularisierung* sowie *Dropout* verknüpft, um Overfitting vorzubeugen (vgl. Liu, 2025, S.40f & 194).\n",
```
Comment (LBeissner) [inline notebooks/fft_net.ipynb]: Bitte klar trennen, was von Mey et al. ist und was von uns. Statt:

Im Fall eines Fully Connected Netzes werden, analog zu Mey et al (2020), 2048 Neuronen in der ersten Dense-Schicht verwendet und durch *L2-Regularisierung* sowie *Dropout* verknüpft, um Overfitting vorzubeugen.

eher:

Wie bei Mey et al. umfasst das erste Dense Layer 2048 Neuronen. In diesem Projekt ist dieses Layer um eine *L2-Regularisierung* und ein Dropout Layer ergänzt, um Overfitting vorzubeugen.

Comment (LBeissner) [inline notebooks/fft_net.ipynb]: Der Part mit Convolutions wird wahrscheinlich wegfallen.

File: notebooks/fft_net.ipynb [hunk 2]
```diff
@@ -149,11 +197,22 @@
...
+    "\n",
+    "Als Aktivierungsfunktion kommt eine *LeakyReLU* mit einstellbarem *negative_slope* zum Einsatz, um das sogenannte *dying ReLU*‑Problem, also die langfristige Deaktivierung einzelner Neuronen, zu vermeiden (Adari & Alla, 2024, S.200). Abschließend wird im Output-Layer eine *Softmax*-Aktivierungsfunktion genutzt, um damit das Modell auf eine Multiklassen-Klassifikation anzupassen. Die Softmax-Funktion wird für Klassifikationsaufgaben verwendet, wo eine spezifische Klasse aus dem Input verhergesagt werden soll (Adari & Alla, 2024, S.203)."
```
Comment (LBeissner) [inline notebooks/fft_net.ipynb]: *vorhergesagt

File: notebooks/fft_net.ipynb [hunk 3]
```diff
@@ -169,14 +228,41 @@
     ")"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Im anschließenden Schritt wird das zuvor definierte Modell mithilfe der Funktion `compile` für das Training vorbereitet. Über den Parameter `learning_rate` wird festgelegt, wie stark die Gewichte des Netzes bei jedem Update angepasst werden (vgl. Adari & Alla, 2024, S.216), \n"
```
Comment (LBeissner) [inline notebooks/fft_net.ipynb]: Punkt am Ende

File: notebooks/fft_net.ipynb [hunk 4]
```diff
@@ -194,6 +280,19 @@
...
+    "Precision gibt den prozentualen Anteil korrekter Positivvorhersagen am Gesamtvolumen an, während Recall den Prozentsatz tatsächlich erkannter Positiver innerhalb aller Positiven beschreibt (Adari & Alla, 2024, S.116).\n",
+    "\n",
+    "Adari und Alla definieren die F1-Metrik als harmonischen Mittelwert von Precision und Recall, bei dem ein höherer Wert darauf hinweist, dass beide Größen vergleichsweise hoch sind (2024, S. 116). Liu beschreibt, dass ein steigender F1-Wert auf eine höhere Prognosekraft des Modells schließen lässt: Ein Wert nahe 1 deutet auf ein nahezu perfektes Klassifikationsmodell hin, während ein F1-Wert nahe 0 auf eine sehr geringe Vorhersagefähigkeit schließen lässt (2025, S. 414). Mit dem Modell dieses Projektes wird nach dem Training mit ``X`` Epochen ein Accuracy-Wert von `X` und ein F1-Wert von `X` erreicht. Dies zeigt, dass ``...``."
```
Comment (LBeissner) [inline notebooks/fft_net.ipynb]: Diese zeigt, dass ...

Was wird hier gezeigt? Kann ich nicht lesen.

File: notebooks/fft_net.ipynb [hunk 5]
```diff
@@ -245,7 +367,8 @@
    "metadata": {},
    "source": [
     "### Confusion Matrix\n",
-    "Die Confusion Matrix zeigt, wie gut dein Modell zwischen den verschiedenen Klassen unterscheidet. Jede Zelle stellt die Anzahl der Samples dar, die einer bestimmten Klasse zugeordnet wurden (wahre Labels) und wie das Modell diese Klasse vorhergesagt hat (vorhergesagte Labels).\n",
+    "Um die Qualität der Klassifikationsergebnisse zu prüfen wird abschließend eine Confusion Matrix erstellt. Diese baut auf den predictions der Keras `predict` Methode auf, welche Vorhersagen für die Testdaten generiert (Keras, o.D.-b). Die gleichnamige Wrapperfunktion gibt die tatsächlichen und die vorhergesagten Klassen zurück. Die Funktion `plot_confusion_matrix` erstellt eine Matrix mit den tatsächlichen Klassen in den Zeilen und den vorhergesagten Klassen in den Spalten. Auf der Diagonalen von ... nach ... sind die korrekten Vorhersagen abzulesen. Je dunkler die Farbe der Kachel ist, desto mehr Vorhersagen wurden korrekt getroffen.\n",
```
Comment (LBeissner) [inline notebooks/fft_net.ipynb]: von ... nach ...

Fehlt hier was?

**Review by LBeissner — CHANGES_REQUESTED** [review]

File: notebooks/fft_net.ipynb (hunk 1 above)
Comment (ninakuhlen) [reply notebooks/fft_net.ipynb]: soll das schon raus?

**Review by LBeissner — APPROVED** [review]

**Review by LBeissner — APPROVED** [review]
